csharp/src/Microsoft.ML.OnnxRuntime/CheckpointState.shared.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    ///  Sets various runtime options. 
+    /// </summary>
+    public class CheckpointState : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        /// <summary>
+        /// Default __ctor. Creates default RuntimeOptions
+        /// </summary>
+        public CheckpointState()
+            : base(IntPtr.Zero, true)
+        {
+            var envHandle = OrtEnv.Handle; // just so it is initialized
+        }
+
+        /// <summary>
+        /// Overrides SafeHandle.IsInvalid
+        /// </summary>
+        /// <value>returns true if handle is equal to Zero</value>
+        public override bool IsInvalid { get { return handle == IntPtr.Zero; } }
+
+        public void LoadCheckpoint(string checkpointPath)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtLoadCheckpoint(NativeMethods.GetPlatformSerializedString(checkpointPath), out handle));
```

Comment:
>checkpointPath [](http://example.com/codeflow?start=116&length=14)

I think all the native C-API methods we have require UTF-8 strings? is this not the case here?
If not, need to be fixed to be UTF-8. Otherwise, the string requires conversion to UTF-8.